python/lsst/verify/yamlpersistance.py

```diff
@@ -0,0 +1,176 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Shared code for persisting verify objects to YAML.
+
+The YAML code is centralized in one module to simplify registration code.
+"""
+
+__all__ = []  # code defined by this module only called indirectly
+
+
+import astropy.units as u
+import yaml
+
+from .measurement import Datum, Blob, Measurement
+
+
+def _getValidLoaders():
+    """Return a list of supported YAML loaders.
+
+    For YAML >= 5.1 need a different Loader for the constructor
+
+    Returns
+    -------
+    loaderList : `sequence`
+        A list of loaders that are supported by the current PyYAML version.
+    """
+    loaderList = [yaml.Loader, yaml.CLoader]
+    try:
+        loaderList.append(yaml.FullLoader)
+    except AttributeError:
+        pass
+    try:
+        loaderList.append(yaml.UnsafeLoader)
+    except AttributeError:
+        pass
+    try:
+        loaderList.append(yaml.SafeLoader)
+    except AttributeError:
+        pass
+    return loaderList
+
+
+def _registerTypes():
+    yaml.add_representer(Datum, datum_representer)
+    yaml.add_representer(Blob, blob_representer)
+    yaml.add_representer(Measurement, measurement_representer)
+
+    for loader in _getValidLoaders():
+        yaml.add_constructor(
+            "lsst.verify.Datum", datum_constructor, Loader=loader)
+        yaml.add_constructor(
+            "lsst.verify.Blob", blob_constructor, Loader=loader)
+        yaml.add_constructor(
+            "lsst.verify.Measurement", measurement_constructor, Loader=loader)
+
+
+# Based on Measurement.json, but provides self-contained representation
+def measurement_representer(dumper, measurement):
+    """Persist a Measurement as a mapping.
+    """
+    if measurement.quantity is None:
+        normalized_value = None
+        normalized_unit_str = None
+    elif measurement.metric is not None:
+        # ensure metrics are normalized to metric definition's units
+        metric = measurement.metric
+        normalized_value = measurement.quantity.to(metric.unit).value
+        normalized_unit_str = metric.unit_str
+    else:
+        normalized_value = measurement.quantity.value
+        normalized_unit_str = str(measurement.quantity.unit)
+
+    return dumper.represent_mapping(
+        "lsst.verify.Measurement",
+        {"metric": str(measurement.metric_name),
+         "identifier": measurement.identifier,
+         "value": normalized_value,
+         "unit": normalized_unit_str,
+         "notes": dict(measurement.notes),
+         # extras included in blobs
+         "blobs": [blob for _, blob in measurement.blobs.items()],
```

Comment:
```python
"blobs": list(measurement.blobs.values()),
```
?

Reply:
Unfortunately, `BlobSet` does not have a `values` method. `verify` is full of classes that superficially resemble standard collections but can't be used like them. :angry:

Reply:
can you add one? ;-)